public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/ResourcePicker.test.tsx

```diff
@@ -125,28 +120,30 @@ describe('AzureMonitor ResourcePicker', () => {
     expect(onApply).toBeCalledWith('/subscriptions/def-123');
   });
 
-  it('should call onApply with a template variable when a user selects it', async () => {
```

Comment:
I'm not sure the current template variables ui makes sense if you're using multiple template variables though? If you had a template variable it would still show in the advanced section?

Reply:
no, this would be for the previously supported case of using just one template variable (and that single template variable is what it would be shown in the Advanced section). Does that make sense?

Reply:
If they previously had just one template variable it would show in the advanced section. Is that not enough? I guess it doesn't hurt to bring it back! Can always remove it later if we think the UI is too cluttered. I guess I can see an argument that if you did have one template variable it might be annoying to type and so you'd rather just click on it.

Reply:
eh ok. So I did experiment with bringing this back but I landed on feeling like it was more confusing than helpful. When we list the possible template variables to choose from, we make it seem like they are all valid options, when they might not be. If you want to use multiple template variable it can seem like you should be able to select multiple variables by clicking many checkboxes but this is not possible. 

I think your suggestion to keep advanced auto expanded when it detects that a user has a template variable will make it clear how to use it:
Here is what a user should see if they are currently using the workspaces() template variable for example
<img width="998" alt="Screen Shot 2022-03-24 at 9 40 37 PM" src="https://user-images.githubusercontent.com/6620164/160037839-3cabe824-856d-4d72-8eb1-e2b8c270c680.png">
<img width="556" alt="Screen Shot 2022-03-24 at 9 40 44 PM" src="https://user-images.githubusercontent.com/6620164/160037840-153278ec-b68b-44e8-af74-23464dc505c8.png">

Reply:
fair enough! we lose the ability of listing the variables but it's clear / easy enough. Thanks for checking!